src/tm_mad/fs_lvm/mv

```diff
@@ -77,22 +78,37 @@ if [ `is_disk $DST_PATH` -eq 1 ]; then
     while IFS= read -r -d '' element; do
         XPATH_ELEMENTS[i++]="$element"
     done < <(onevm show -x $VMID | $XPATH  \
-                        /VM/TEMPLATE/DISK[DISK_ID=$DISK_ID]/TYPE )
+                        /VM/LCM_STATE )
 
-    TYPE="${XPATH_ELEMENTS[j++]}"
-
-    ####
-
-    if [ "${TYPE}" != "BLOCK" ]; then
```

Comment:
I don't understand this yet. It still looks to me like can-not-happen case, once it passed the condition on line 67 (is_disk) it should always be "BLOCK", right?

Reply:
I presume this check was added there for the case if datastore was converted from `shared` to `fs_lvm`, as described in [documentation](http://docs.opennebula.org/5.8/deployment/open_cloud_storage_setup/lvm_drivers.html):
> It is a good idea to first deploy a shared FileSystem datastore and once it is working replace the associated System datastore with the LVM one maintaining the shared mount point as described below.

There two things about that:

1) Check is not working, because `/VM/TEMPLATE/DISK[DISK_ID=$DISK_ID]/TYPE` always equals `FILE`, I presume it might be working if we would replace it to `DISK_TYPE` which is inherit from the datastore, but in this case we lose opportunity to change it on the fly. Once this parameter set for the image it can't be updated in the future.
2) Better to check it on the place, because it also adds opportunity for convert simple file to the block device by simple migrating datastore via sunstone.
https://github.com/OpenNebula/one/blob/5a4e6633223fdd67c5ac0205c41339b559fb864c/src/tm_mad/fs_lvm/delete#L57-L58